File: surf-friends/src/main/java/dev/slne/surf/friends/command/FriendCommand.java
package dev.slne.surf.friends.command;

import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.surf.friends.command.subcommand.FriendAcceptCommand;
import dev.slne.surf.friends.command.subcommand.FriendAddCommand;
import dev.slne.surf.friends.command.subcommand.FriendDenyCommand;
import dev.slne.surf.friends.command.subcommand.FriendJumpCommand;
import dev.slne.surf.friends.command.subcommand.FriendRemoveCommand;
import dev.slne.surf.friends.command.subcommand.FriendToggleCommand;

public class FriendCommand extends CommandAPICommand {

  public FriendCommand(String commandName) {
    super(commandName);

    withPermission("surf.friends.command.friend");

    withSubcommand(new FriendAcceptCommand("accept"));
    withSubcommand(new FriendAddCommand("add"));
    withSubcommand(new FriendDenyCommand("deny"));
    withSubcommand(new FriendRemoveCommand("remove"));
    withSubcommand(new FriendToggleCommand("toggle"));
    withSubcommand(new FriendJumpCommand("jump"));

  }
}

File: surf-friends/src/main/java/dev/slne/surf/friends/command/FriendCommand.kt
package dev.slne.surf.friends.command

import dev.jorel.commandapi.CommandAPI
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.slne.surf.friends.command.subcommand.*
import dev.slne.surf.friends.menu.FriendMainMenu
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player

class FriendCommand(commandName: String) : CommandAPICommand(commandName) {
    
    init {
        withPermission("surf.friends.command.friend")

        withSubcommand(FriendAcceptCommand("accept"))
        withSubcommand(FriendAddCommand("add"))
        withSubcommand(FriendDenyCommand("deny"))
        withSubcommand(FriendRemoveCommand("remove"))
        withSubcommand(FriendToggleCommand("toggle"))
        withSubcommand(FriendJumpCommand("jump"))
        withSubcommand(FriendListCommand("list"))
        withSubcommand(FriendRequestListCommand("requests"))

        executesPlayer(PlayerCommandExecutor { player: Player, _: CommandArguments? ->
            FriendMainMenu().show(player)
        })
    }
}

fun <T> CommandArguments.getUncheckedOrFail(
    nodeName: String,
    message: String? = "$nodeName not found"
): T = getUnchecked<T>(nodeName) ?: throw CommandAPI.failWithString(message)

fun CommandArguments.getOfflinePlayerOrFail(nodeName: String): OfflinePlayer = getUncheckedOrFail(nodeName, "Der Spieler wurde nicht gefunden.")